src/components/Composer/index.js

```diff
@@ -122,380 +123,352 @@ const IMAGE_EXTENSIONS = {
-        // get a ref to the inner textInput element e.g. if we do
-        // <constructor ref={el => this.textInput = el} /> this will not
-        // return a ref to the component, but rather the HTML element by default
-        if (this.props.forwardedRef && _.isFunction(this.props.forwardedRef)) {
-            this.props.forwardedRef(this.textInput);
-        }
-
-        // There is no onPaste or onDrag for TextInput in react-native so we will add event
-        // listeners here and unbind when the component unmounts
-        if (this.textInput) {
-            this.textInput.addEventListener('wheel', this.handleWheel);
-
-            // we need to handle listeners on navigation focus/blur as Composer is not unmounting
-            // when navigating away to different report
-            this.unsubscribeFocus = this.props.navigation.addListener('focus', () => document.addEventListener('paste', this.handlePaste));
-            this.unsubscribeBlur = this.props.navigation.addListener('blur', () => document.removeEventListener('paste', this.handlePaste));
+    // Find the index of the next space or new line character
+    const spaceIndex = substr.search(/[ \n]/);
 
-            // We need to add paste listener manually as well as navigation focus event is not triggered on component mount
-            document.addEventListener('paste', this.handlePaste);
-        }
+    if (spaceIndex === -1) {
+        return substr;
     }
 
-    componentDidUpdate(prevProps) {
-        if (!prevProps.shouldClear && this.props.shouldClear) {
-            this.textInput.clear();
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({numberOfLines: 1});
-            this.props.onClear();
-        }
-
-        if (
-            prevProps.value !== this.props.value ||
-            prevProps.defaultValue !== this.props.defaultValue ||
-            prevProps.isComposerFullSize !== this.props.isComposerFullSize ||
-            prevProps.windowWidth !== this.props.windowWidth ||
-            prevProps.numberOfLines !== this.props.numberOfLines
-        ) {
-            this.updateNumberOfLines();
-        }
-
-        if (prevProps.selection !== this.props.selection) {
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({selection: this.props.selection});
-        }
-    }
+    // If there is a space or new line, return the substring up to the space or new line
+    return substr.substring(0, spaceIndex);
+};
 
-    componentWillUnmount() {
-        if (!this.textInput) {
+// Enable Markdown parsing.
+// On web we like to have the Text Input field always focused so the user can easily type a new chat
+function Composer({
+    onKeyPress,
+    style,
+    shouldClear,
+    onClear,
+    checkComposerVisibility,
+    onPasteFile,
+    translate,
+    isComposerFullSize,
+    maxLines,
+    value,
+    onNumberOfLinesChange,
```

Comment:
Please move all the default values here. Right now we are not defining defaultValues like in flow.
```suggestion
    onNumberOfLinesChange: () => {},
```

Reply:
for reference: https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring

Reply:
I've reviewed the main branch, and there are still rules in place that require default values. When I remove these default values, the rules flag this as an error, indicating that defaultValues are mandatory

Reply:
Sorry for the misunderstanding, looks like this is prepared for TS migration, https://expensify.slack.com/archives/C01GTK53T8Q/p1691052728177889